### PR TITLE
Compact flag feature

### PR DESCRIPTION
**Pull request by psteyer**
Added a feature to simplify linter output.

Related issue: https://github.com/anaconda-distribution/anaconda-linter/issues/79

### CODE REVIEW

File: anaconda_linter/lint/__init__.py [hunk 1]
```diff
@@ -546,6 +546,11 @@ def get_report(
                     \n\t{msg.body}"
                 for msg in messages
             )
+        elif compact:
+            name = messages[0].recipe.meta.get("package", {}).get("name", None)
+            version = messages[0].recipe.meta.get("package", {}).get("version", None)
+            return f"{name} - {version}\n" + "\n".join(f"\t{msg.title}" for msg in messages)
```
Comment (marcoesters) [inline anaconda_linter/lint/__init__.py:552]: Tab characters are implemented differently different operating systems, so I would avoid using them.

**Comment by marcoesters** [comment]
This implementation is missing a guard against using both `verbose` and `compact` at the same time. We shouldn't be allowed to do this.

`verbose` may take precedence, but there is too high of a risk that there may be interference in future implementations.

We may also want to consider using some type of enum or integer constants to set the output level instead of adding an additional bool to the function input. If we are 100% sure that this "verbose" and "compact" are the end of it, this may be overkill. But are we sure?

File: anaconda_linter/lint/__init__.py (hunk 1 above)
Comment (marcoesters) [reply anaconda_linter/lint/__init__.py:552]: I do not see the change. It still says "\t" in the code for me.

Comment (ous8292) [reply anaconda_linter/lint/__init__.py:552]: I was doing a little research about this and came across the following link
https://docs.python.org/3/reference/lexical_analysis.html#string-and-bytes-literals

It looks like `\t` is an ASCII Horizontal Tab. So python should handle converting that into a tab for us right?

Comment (marcoesters) [reply anaconda_linter/lint/__init__.py:552]: Yes, it's perfectly capable of doing that. However, keep in mind that if the compact output will be the default output, you have to make this easy to parse. A tab character is indistinguishable in display from space characters and it just makes it more annoying to deal with when parsing, in my opinion.

The more I think about it, the more I think we should just remove the leading indent. I don't think it provides much value.

**Comment by marcoesters** [comment]
Closing due to inactivity - we can revitalize it when the need arises

**Pull request closed without merging**